### PR TITLE
BibAuthorID: fix for empty list of authors in search query

### PR DESCRIPTION
**Pull request by MSusik**
Fixes https://sentry.cern.ch/inspire-sentry/inspire/group/359/

Signed-off-by: Mateusz Susik mateusz.susik@cern.ch

### CODE REVIEW

File: modules/bibauthorid/lib/bibauthorid_dbinterface.py [hunk 1]
```diff
@@ -3737,11 +3737,11 @@ def get_name_variants_for_authors(authors):  # get_indexable_name_personids
...
-    name_variants = run_sql("""select id, personids
-                               from aidDENSEINDEX
-                               where id in %s
-                               and flag=1"""
-                            % _get_sqlstr_from_set(authors))
+
+    name_variants = run_sql("""select id, personids from aidDENSEINDEX
+                               where%s flag=1"""
+                            % (" id in " + _get_sqlstr_from_set(authors) + " and"
+                               if len(authors) else ""))
```
Comment (MSusik) [inline modules/bibauthorid/lib/bibauthorid_dbinterface.py:3744]: Two cases:
`select id, personids from aidDENSEINDEX where flag=1`
or
`select id, personids from aidDENSEINDEX where id in ? and flag=1`

File: modules/bibauthorid/lib/bibauthorid_dbinterface.py [hunk 2]
```diff
@@ -3737,11 +3737,11 @@ def get_name_variants_for_authors(authors):  # get_indexable_name_personids
     @return: real author name and the author identifiers which carry that name ((name, pids),)
     @rtype: tuple ((str, bytes),)
     '''
-    name_variants = run_sql("""select id, personids
-                               from aidDENSEINDEX
-                               where id in %s
-                               and flag=1"""
-                            % _get_sqlstr_from_set(authors))
+
```
Comment (MSusik) [inline modules/bibauthorid/lib/bibauthorid_dbinterface.py:3740]: Returns all the authors in the mentioned case

File: modules/bibauthorid/lib/bibauthorid_dbinterface.py (hunk 1 above)
Comment (kaplun) [inline modules/bibauthorid/lib/bibauthorid_dbinterface.py:3744]: I rather think this can be explosive so rather prefer the other implementation :wink:

Comment (kaplun) [inline modules/bibauthorid/lib/bibauthorid_dbinterface.py:3744]: 941344 lines returned on production

**Comment by kaplun** [comment]
Taken the other branch.

**Pull request closed without merging**